### DISCUSSION TITLE
Hardware station is searching for the Third-party dll's in different path for win 10 and win11

### DISCUSSION THREAD

**Discussion by LavanyaNiki** [microsoft/Dynamics365Commerce.Solutions#201] [Q&A] [answered]
We have integrated third party payment connector in the Store Commerce. In the third-party payment connector, we have x86, x64 architecture based dll's.

below is the screen shot for the payment connector dll's ,
1) c3agent 
2) other dll's are in x64 and x86 folder each folder contains 7 dll, they are same dll only differs in architecture.

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/4e4ed05d-17c2-4332-bc5a-c2c60471639b)

Some part of code in c3 agent dll :

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/313aab54-f571-42cf-8155-202778cfada2)

Information on how C3Agent C# is loading the PCL .dlls :
•	First It calls SetDllDirectory of kernel32.dll in order to add the PCL directory to the system search path used to locate DLLs for the application
•	Depending on the target machine (32 or 64 bits), C3A calls SetDllDirectory with ${CWD}\x86 or ${CWD}\x64 (${CWD} : current working directory of the POS application)

During the execution we are seeing the Hardware station trying to load/capture the third-party payment connector DLL's in different directory based on Win 11 and Win 10 machines.

We have used process monitor tool to understanding the DLL loading path. Here are the result what we got:
win10:
In Windows 10, it primarily searches in a different location initially.

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/6280b483-b9ec-45f7-8d32-4f4fbeda95b4)

win 11:
In Windows 11, it primarily searches in a different location initially.

![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/140384016/f768fd34-06a7-4aea-80c2-c7f5f188a534)

So, we have following questions:
 1) We want to understand what is the working directory of the Store Commerce application?
 2) Why the setup is behaving differently on windows 10 and windows 11?

**Comment by madyke** [comment, marked as answer]
Hello @LavanyaNiki - for your first question about the working directory that Store Commerce uses to look for DLLs, it will first check:
`C:\Program files\Microsoft Dynamics 365\10.0\Store Commerce\Extensions\<extensionName>\HardwareStation`
if it cannot find DLLs in that path then it will default to:
`C:\Program files\Microsoft Dynamics 365\10.0\Store Commerce\Microsoft\contentfiles\HardwareStation\`.

If the application cannot find the DLLs in either of those locations, then Windows will search the other folders listed in the system search path, which are all the other calls you're seeing in process monitor. Unfortunately, since that isn't Dynamics code we can't really provide an answer for your second question - you'll either need to work with the third-party provider or review the APIs used in their code yourself to understand why they are behaving differently on Windows 10/11.
